docs/src/pages/components/data-grid/columns/ColumnOrderingGrid.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { XGrid } from '@material-ui/x-grid';
+
+const rows = [
+  {
+    id: 1,
+    username: 'defunkt',
+    age: 38,
+  },
+];
+
+export default function ColumnWidthGrid() {
+  return (
+    <div style={{ height: 250, width: '100%' }}>
+      <XGrid
+        columns={[
+          { field: 'id' },
+          { field: 'username' },
+          { field: 'age', resizable: false },
+        ]}
+        rows={rows}
```

Comment:
What about we demo a more complex case?

Maybe the same case as https://github.com/mui-org/material-ui-x/blob/74d4fc9fbfacfc154b10a98a2e99e74596107bc5/docs/src/pages/components/data-grid/pagination/200PaginationGrid.tsx#L9

but with more columns to see how it behaves with horizontal scrolling?

Reply:
> We are getting closer :). Regarding the 6 above issues:
> 
> 1. "Infinite loop" is partially solved. We can still see two columns flip orders multiple times when trying to swap two.
> 2. "Broken checkbox" seems to be solved
> 3. "Hope to resize but drag instead" this will likely need to solve #209 first
> 4. "Sorting after reordering crashes" seems to be solved
> 5. "Can't reorder when moving outside the grid" I guess we can solve that later on. It imagines it will require to drop the native drag & drop API. Once we do that, we should also be able to use drag & drop to allow end-users to pin columns. It's similar to #191.
> 6. "Wrong column position" This one is my biggest concern. The position of the column is frequently not correct. It makes the feature no usable in practice.
>    Sometimes there is an offset. You can observe a similar issue with the threshold for moving columns.
>    Sometimes it doesn't trigger once the cursor crosses a column boundary.
> 
> In terms of future features, I can think of the following (that we will definitely not cover here):
> 
> * Transition when swapping
> * Disabling reorder on one column
> * Fixing the position of one column (stronger that disabling reorder on one column)
> * Column group reorder
> 
> We can open new issues once we are done with the first iteration.

First about the 3 and 6 points -> to be honest, I wast able to reproduce nither of them

For 5 -> yes, that one is a bit more tricky and will require basically the same exact solution as the column resize feature because the `useColumnResize` suffers from the same problem. This doesn't necessarily mean that we need to drop the native DnD HTML5 API but just have a more complete solution of detecting cursor movement and position inside the grid.

For the follow-up features -> let's include this in the next team enterprise meeting and create a follow-up ticket or tickets that contain those features so that we can plan when to take them.

And last but not least -> I'll check now how to fully resolve the 1st point.

PS: I'll also add a more complex example for the docs